src/prefect/utilities/logging.py

```diff
@@ -207,6 +266,9 @@ def _create_logger(name: str) -> logging.Logger:
     cloud_handler = CloudHandler()
     cloud_handler.setLevel("DEBUG")
     logger.addHandler(cloud_handler)
+
+    _add_redactor()
```

Comment:
We are considering whether the check for whether this will run anything should be inline here (easier to reason that it will be skipped without jumping to the function for new readers) or if it is cleaner the way that it is. I will double check for precedent elsewhere in the codebase, but fundamentally I think this is a matter of style.